client/python/tests/unit/test_client.py

```diff
@@ -161,7 +161,15 @@ def test_get_queue_info():
 def test_cancel_jobs():
     test_create_queue()
     test_submit_job()
-    tester.cancel_jobs(queue="test", job_set_id="job-set-1", job_id="job1")
+
+    # Test that the right combination of jobid or jobsetid and queue is used
+    # also check that the Value error is raised
+    with pytest.raises(ValueError):
+        tester.cancel_jobs(queue="test", job_id="job-1", job_set_id="job-set-1")
+
+    tester.cancel_jobs(job_id="job-1")
```

Comment:
Wouldn’t the first call cancel the job and the second wouldn’t be a no opp?

when writing tests, it’s usually good to follow S
RP. You don’t want side effects of one test influencing another.

Reply:
Oh yeah. I didn't think about this. I will fix it.

Reply:
In hindsight, I do not think this applies for the unit tests - these are only testing that the gRPC server accepts the request as valid - there is no armada behavior checks and therefore no way to test if the cancelation is successful.

Reply:
You are calling a mock for these unit tests. Nothing is stopping you from implementing a test cancel function

Reply:
You also could check returns here.

Reply:
Yeah okay - I am just confused why we didn't do this earlier. I will redo the mock class to support full return statement checking across the board